tests/src/Gui/Camera.cpp

```diff
@@ -0,0 +1,220 @@
+    SbVec3f vo, vx, vy, vz;
+    matrix.multVecMatrix(SbVec3f(0, 0, 0), vo);
+    matrix.multVecMatrix(SbVec3f(10, 0, 0), vx);
+    matrix.multVecMatrix(SbVec3f(0, 10, 0), vy);
+    matrix.multVecMatrix(SbVec3f(0, 0, 10), vz);
+
+    // Project to XY plane by setting Z to 0
+    vo[2] = 0;
+    vx[2] = 0;
+    vy[2] = 0;
+    vz[2] = 0;
+
+    // Return the lengths of the projected vectors
+    return {(vx - vo).length(), (vy - vo).length(), (vz - vo).length()};
+}
+
+}  // namespace
+
+TEST(CameraPrecalculatedQuaternions, testIsometric)
+{
+    // Use the formula to get the isometric rotation
+    double alpha = toRadians(45.0f);
+    double beta = std::asin(-std::sqrt(1.0 / 3.0));
+
+    const Rotation actual = buildAxonometricRotation(alpha, beta);
+    const Rotation expected = toRotation(Gui::Camera::isometric());
+
+    EXPECT_TRUE(actual.isSame(expected, 1e-6));
+}
+
+TEST(CameraPrecalculatedQuaternions, testDimetric)
+{
+    // Use the formula to get the dimetric rotation
+    double alpha = std::asin(std::sqrt(1.0 / 8.0));
+    double beta = -std::asin(1.0 / 3.0);
+
+    const Rotation actual = buildAxonometricRotation(alpha, beta);
+    const Rotation expected = toRotation(Gui::Camera::dimetric());
+
+    EXPECT_TRUE(actual.isSame(expected, 1e-6));
+}
+
+TEST(CameraPrecalculatedQuaternions, testTrimetric)
+{
+    // Use the formula to get the trimetric rotation
+    double alpha = toRadians(30.0);
+    double beta = toRadians(-35.0);
+
+    const Rotation actual = buildAxonometricRotation(alpha, beta);
+    const Rotation expected = toRotation(Gui::Camera::trimetric());
+
+    EXPECT_TRUE(actual.isSame(expected, 1e-6));
+}
+
+TEST(CameraRotation, testIsometricProjection)
+{
+    auto rot = Gui::Camera::isometric();
+    auto lengths = getProjectedLengths(rot);
+
+    // In isometric, expect all lengths to be roughly equal
+    EXPECT_NEAR(lengths[0], lengths[1], 1e-6);  // X == Y
+    EXPECT_NEAR(lengths[0], lengths[2], 1e-6);  // X == Z
+    EXPECT_NEAR(lengths[1], lengths[2], 1e-6);  // Y == Z
+}
+
+TEST(CameraRotation, testDimetricProjection)
+{
+    auto rot = Gui::Camera::dimetric();
+    auto lengths = getProjectedLengths(rot);
+
+    // In dimetric, expect two lengths to be roughly equal, one different
+    int similarCount = 0;
+    if (std::abs(lengths[0] - lengths[1]) < 1e-6) {
+        similarCount++;
+    }
+    if (std::abs(lengths[1] - lengths[2]) < 1e-6) {
+        similarCount++;
+    }
+    if (std::abs(lengths[0] - lengths[2]) < 1e-6) {
+        similarCount++;
+    }
+
+    EXPECT_EQ(similarCount, 1);  // Exactly two are equal
+}
+
+TEST(CameraRotation, testTrimetricProjection)
+{
+    auto rot = Gui::Camera::trimetric();
+    auto lengths = getProjectedLengths(rot);
+
+    // In trimetric, all should differ significantly
+    EXPECT_GT(std::abs(lengths[0] - lengths[1]), 1e-3);
+    EXPECT_GT(std::abs(lengths[1] - lengths[2]), 1e-3);
+    EXPECT_GT(std::abs(lengths[0] - lengths[2]), 1e-3);
```

Comment:
`EXPECT_NEAR` would not be better in that case?

Reply:
In this test I'm checking that the projected lengths differs from each other (see comment above in code).
Testing with `near` would mean that the lengths are close, which is opposite to what I want to do.
Or am I misunderstanding what you mean?

Reply:
Oh, you are right I misread the condition.

I'd expect GTest to have negation of EXPECT_NEAR but it is not the case... See https://github.com/google/googletest/issues/623. They suggest to use `ASSERT_THAT` with composable marchers but I don't think that it is worth here.